centreon/src/Core/Dashboard/Infrastructure/Repository/DbReadDashboardShareRepository.php

```diff
@@ -361,6 +364,275 @@ public function findDashboardsContactGroupSharesByContact(ContactInterface $cont
+                FROM `:db`.contact c
+                    LEFT JOIN `:db`.contactgroup_contact_relation cgcr
+                        ON cgcr.contact_contact_id = c.contact_id
+                    LEFT JOIN `:db`.acl_group_contactgroups_relations gcgr
+                        ON gcgr.cg_cg_id = cgcr.contactgroup_cg_id
+                    LEFT JOIN `:db`.acl_group_contacts_relations gcr
+                        ON gcr.contact_contact_id = c.contact_id
+                    LEFT JOIN `:db`.acl_group_topology_relations agtr
+                        ON agtr.acl_group_id = gcr.acl_group_id
+                            OR agtr.acl_group_id = gcgr.acl_group_id
+                    LEFT JOIN `:db`.acl_topology_relations acltr
+                        ON acltr.acl_topo_id = agtr.acl_topology_id
+                    INNER JOIN `:db`.topology
+                        ON topology.topology_id = acltr.topology_topology_id
+            SQL;
+
+        $searchRequest = $sqlTranslator->translateSearchParameterToSql();
+        $query .= $searchRequest !== null
+            ? $searchRequest . ' AND '
+            : ' WHERE ';
+
+        $query .= <<<SQL
+            topology.topology_page IN (10401,10402,10403)
+                AND gcr.acl_group_id IN ({$bindTokenAsString})
+                AND acltr.access_right IS NOT NULL
+                AND c.contact_oreon = '1'
+            GROUP BY c.contact_id
+            SQL;
+
+        $statement = $this->db->prepare($this->translateDbName($query));
+        foreach ($bind as $token => $aclGroupId) {
+            $statement->bindValue($token, $aclGroupId, \PDO::PARAM_INT);
+        }
+        foreach ($sqlTranslator->getSearchValues() as $key => $data) {
+            /**
+             * @var int
+             */
+            $type = key($data);
+            $value = $data[$type];
+            $statement->bindValue($key, $value, $type);
+        }
+        $statement->execute();
+
+        $result = $this->db->query('SELECT FOUND_ROWS()');
+        if ($result !== false && ($total = $result->fetchColumn()) !== false) {
+            $sqlTranslator->getRequestParameters()->setTotal((int) $total);
+        }
+
+        $dashboardContactRoles = [];
+        foreach ($statement as $contactRole) {
+            /** @var array{
+             *     topologies: string,
+             *     contact_name: string,
+             *     contact_id: int,
+             *     contact_email: string
+             * } $contactRole
+             */
+            $dashboardContactRoles[] = $this->createDashboardContactRole($contactRole);
+        }
+
+        return $dashboardContactRoles;
+    }
+
+    /**
+     * @param RequestParametersInterface $requestParameters
+     *
+     * @throws \Throwable
+     *
+     * @return DashboardContactRole[]
+     */
+    private function findDashboardAdminWithRequestParameters(RequestParametersInterface $requestParameters): array
```

Comment:
Admin is a business context. You shouldn't create repository methods with business context notions.